apps/src/StudioApp.js

```diff
@@ -757,6 +767,82 @@ StudioApp.prototype.onResize = function() {
   this.resizeToolboxHeader();
 };
 
+StudioApp.prototype.onMouseDownVizResizeBar = function (event) {
+  // When we see a mouse down in the resize bar, start tracking mouse moves:
+
+  this.onMouseMoveBoundHandler = _.bind(this.onMouseMoveVizResizeBar, this);
+  document.body.addEventListener('mousemove', this.onMouseMoveBoundHandler);
+
+  event.preventDefault();
+};
+
+function applyTransformScaleToChildren(element, scale) {
+  for (var i = 0; i < element.children.length; i++) {
+    element.children[i].style.transform = scale;
+    element.children[i].style.msTransform = scale;
+    element.children[i].style.webkitTransform = scale;
+  }
+}
+
+/**
+*  Handle mouse moves while dragging the visualization resize bar. We set
+*  styles on each of the elements directly, overriding the normal responsive
+*  classes that would typically adjust width and scale.
+*/
+StudioApp.prototype.onMouseMoveVizResizeBar = function (event) {
+  var codeWorkspace = document.getElementById('codeWorkspace');
+  var visualizationResizeBar = document.getElementById('visualizationResizeBar');
+  var visualization = document.getElementById('visualization');
+  var visualizationColumn = document.getElementById('visualizationColumn');
+  var visualizationEditor = document.getElementById('visualizationEditor');
+
+  var rect = visualizationResizeBar.getBoundingClientRect();
+  var offset;
+  var newVizWidth;
+  if (this.isRtl()) {
+    offset = window.innerWidth -
+             (window.scrollX + rect.left + (rect.width / 2)) -
+             parseInt(window.getComputedStyle(visualizationResizeBar).right, 10);
+    newVizWidth = (window.innerWidth - event.pageX) - offset;
+  } else {
+    offset = window.scrollX + rect.left + (rect.width / 2) -
```

Comment:
Does this work in other browsers? I remember cross-browser problems with scrollX. Here's some discussion: https://developer.mozilla.org/en-US/docs/Web/API/Window/scrollX , which suggests "For cross-browser compatibility, use window.pageXOffset instead of window.scrollX." I haven't confirmed this, but may be useful if broken in other browsers